### PR TITLE
DEV-2347 Integrating the new events API

### PR DESCRIPTION
**Pull request by pauldwolfe**
Not much of a functional change as all we really need to do is remove the passthrough, as now we'll only have one event per run. Also remove the shallow context check as shallows will be filtered by ini.

### CODE REVIEW

File: src/main/java/com/hartwig/snpcheck/SnpCheck.java [hunk 1]
```diff
@@ -64,49 +63,45 @@ public SnpCheck(final RunApi runs, final SampleApi samples, final Bucket snpchec
...
-                            Sample refSample = maybeRefSample.get();
-                            Optional<Blob> maybeValVcf = findValidationVcf(valVcfs, refSample);
-                            if (maybeValVcf.isPresent()) {
-                                VcfComparison.Result result = doComparison(run, refSample, maybeValVcf.get());
-                                SnpCheckEvent.builder()
-                                        .publisher(turquoiseTopicPublisher)
-                                        .sample(maybeTumorSample.map(Sample::getName).orElse(refSample.getName()))
-                                        .result(result.name().toLowerCase())
+            Run run = runs.get(event.pipeline().runId());
+            if (run.getIni().equals(Ini.SOMATIC_INI.getValue()) || run.getIni().equals(Ini.SINGLESAMPLE_INI.getValue())) {
```
Comment (nedleitch) [inline src/main/java/com/hartwig/snpcheck/SnpCheck.java:69]: This is slightly different than the original intent, does it make sense from a maintenance perspective just to look for NOT shallow?

File: src/main/java/com/hartwig/snpcheck/SnpCheck.java [hunk 2]
```diff
@@ -64,49 +63,45 @@ public SnpCheck(final RunApi runs, final SampleApi samples, final Bucket snpchec
...
+                LOGGER.info("Received a SnpCheck candidate [{}] for run [{}]", run.getSet().getName(), run.getId());
+                if (waitForFinished(run)) {
```
Comment (nedleitch) [inline src/main/java/com/hartwig/snpcheck/SnpCheck.java:71]: Will we encounter unfinished runs anymore now that all the events are together?

**Review by nedleitch — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 3]
```diff
@@ -24,7 +24,7 @@
         <google-cloud-bom.version>0.145.0</google-cloud-bom.version>
         <java-client.version>1.1.26</java-client.version>
         <pipeline5.version>5.18.1943</pipeline5.version>
-        <pipeline-events.version>1.2.20</pipeline-events.version>
+        <pipeline-events.version>local-SNAPSHOT</pipeline-events.version>
```
Comment (nedleitch) [inline pom.xml:27]: I think you have `2.0.0` now?

File: src/main/java/com/hartwig/snpcheck/SnpCheck.java (hunk 1 above)
Comment (pauldwolfe) [reply src/main/java/com/hartwig/snpcheck/SnpCheck.java:69]: i think the opposite, we don't want to try to snpcheck a Rerun.ini or an Rna.ini either.

File: src/main/java/com/hartwig/snpcheck/SnpCheck.java (hunk 2 above)
Comment (pauldwolfe) [reply src/main/java/com/hartwig/snpcheck/SnpCheck.java:71]: Indeed we can!

**Pull request merged**